uf3/regression/least_squares.py

```diff
@@ -176,11 +185,17 @@ def set_params(self, **params):
             self.bspline_config = params["bspline_config"]
         if "regularizer" in params:
             self.regularizer = params["regularizer"]
+        if "ridge_map" in params:
+            self.ridge_map = params["ridge_map"]
+        if "curvature_map" in params:
+            self.curvature_map = params["curvature_map"]
         elif self.regularizer is None:
             reg_params = {k: v for k, v in params.items()
                           if isinstance(v, (int, float, np.floating))}
-            reg = self.bspline_config.get_regularization_matrix(**reg_params)
+            reg,ridge_map,curvature_map = self.bspline_config.get_regularization_matrix(**reg_params)
```

Comment:
This line is incorrect! bspline_config.get_regularization_matrix returns only one object